modules/3drenderer/Map.js

```diff
@@ -0,0 +1,300 @@
+import { Map as mapLibreMap } from 'maplibre-gl';
+import { STYLES } from '../pixi/styles';
```

Comment:
This is ok for now, but eventually we'll probably let users replace the styles.. 
This is just me thinking out loud, and I don't think you should change anything.

When I did the preset icon code, I ended up just importing the `styleMatch` function which accepts a `tags` object.
https://github.com/facebook/Rapid/blob/main/modules/ui/preset_icon.js#L258

So in your code, I think you could also do `styleMatch({highway: primary})` to get - whatever the primary highway style is now - instead of re-using that `STYLES` const.   Both approaches are kinda hacky and will change later as we figure out #451, so this doesn't really matter. 😆

Reply:
Ah, I remember now. I don't think there's a way for me to refer to the obtained tag value programmatically and pass it into the `styleMatch` method _within the styling expression_, which is what maplibre needs me to do- hence the 1:1 explicit mapping of tag val to color.

Reply:
Update! I realized that I wasn't thinking about things correctly. Instead of passing in the unmodified geojson / tags and then writing dense mapbox styling rules to figure out colors,  INSTEAD it is best to prepare the geojson with whatever custom properties we want, then just have the styling code just read the property.  

TL:DR; during geojson prep, we now call StyleMatch on the entity's tags, grab the fill/stroke/casing color from the returned style object, then add them to the geojson properties:

```
function generateAreaLayer(context, areaEnts, _map) {
....
for (const areaEnt of areaEnts) { 
 ... 
 
 
    const style = styleMatch(areaEnt.tags);
    const fillColor = PIXI.utils.hex2string(style.fill.color);
    const strokeColor = PIXI.utils.hex2string(style.stroke.color);

    let newFeature = {
      type: 'Feature',
      properties: {
        selected: selectedIDs.includes(areaEnt.id).toString(),
        fillcolor: fillColor,
        strokecolor: strokeColor,
      },
      geometry: gj,
    };
```

Then later, in the actual style expression code for the layer: 

```
     paint: {
        'fill-color': ['get', 'fill-Color'],
```

... Much DRYer this way. 
 

THen, the styling rule